src/System.Linq/src/System/Linq/Take.cs

```diff
@@ -96,5 +96,37 @@ private static IEnumerable<TSource> TakeWhileIterator<TSource>(IEnumerable<TSour
                 yield return element;
             }
         }
+
+#if netcoreapp11
+        public static IEnumerable<TSource> TakeLast<TSource>(this IEnumerable<TSource> source, int count)
+        {
+            if (source == null)
+            {
+                throw Error.ArgumentNull(nameof(source));
+            }
+
+            return TakeLastIterator(source, Math.Max(count, 0));
+        }
+
+        private static IEnumerable<TSource> TakeLastIterator<TSource>(IEnumerable<TSource> source, int count)
+        {
+            var queue = new Queue<TSource>();
+
+            foreach (TSource item in source)
+            {
+                if (queue.Count == count)
```

Comment:
Again, once `Count` reaches `count` it won't go below again, so no need to keep checking it.
Also, it will do so after `count` items, so we can use that as our way to detect, rather than a property read.